### ISSUE TITLE
clippy::non-canonical-partial-ord-impl warns if `self.cmp(other).into()` instead of `Some(self.cmp(other))` is used

### ISSUE THREAD

**Issue by Holzhaus** [rust-lang/rust-clippy#13640]
### Summary

The lint causes a warning if `self.cmp(other).into()` is used inside the `PartialEq::partial_cmp()` implementation instead of using `Some(self.cmp(other))`.

### Lint Name

clippy::non-canonical-partial-ord-impl

### Reproducer

I tried this code:

```rust
impl PartialOrd for TaggedFile {
    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
        self.cmp(other).into()
    }
}
```

I saw this happen:

```rust
error: non-canonical implementation of `partial_cmp` on an `Ord` type
   --> src/taggedfile.rs:287:1
    |
287 | /  impl PartialOrd for TaggedFile {
288 | |      fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
    | | _____________________________________________________________-
289 | ||         self.cmp(other).into()
290 | ||     }
    | ||_____- help: change this to: `{ Some(self.cmp(other)) }`
291 | |  }
    | |__^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#non_canonical_partial_ord_impl
    = note: `-D clippy::non-canonical-partial-ord-impl` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::non_canonical_partial_ord_impl)]`
```

I expected to see this happen:

No warningg, because `self.cmp(other).into()` is equivalent to `Some(self.cmp(other))` in this context.

### Version

rustc 1.82.0 (f6e511eec 2024-10-15)
binary: rustc
commit-hash: f6e511eec7342f59a25f7c0534f1dbea00d01b14
commit-date: 2024-10-15
host: x86_64-unknown-linux-gnu
release: 1.82.0
LLVM version: 19.1.1
